Get-ChocolateyVersion fails to get the Chocolatey CLI version when a license is present but the chocolatey.extension package is missing

### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my problem.
- [X] I have verified this is not an issue for a specific package.
- [X] I have verified this issue is not security related.

### What You Are Seeing?

When installing "chocolatey.extenstion" and "chocolatey-core.extension", the playbook is failing with the below error. This is only happening when using the  Chocolatey Ansible Collection v1.5.0. If I specify to use the previous version (v1.4.0), the playbook works fine without any issues.

Error: "exception": "Cannot convert value \"Unable to load chocolatey.licensed assembly. Could not find a part of the path 'C:\\ProgramData\\chocolatey\\extensions\\chocolatey\\chocolatey.licensed.dll'.\r\nA valid Chocolatey license was found, but the chocolatey.licensed.dll assembly could not be loaded:\r\n  Could not load file or assembly 'chocolatey.licensed, Version=0.0.0.0, Culture=neutral, PublicKeyToken=' or one of its dependencies. The system cannot find the file specified.\r\nEnsure that the chocolatey.licensed.dll exists at the following path:\r\n 'C:\\ProgramData\\chocolatey\\extensions\\chocolatey\\chocolatey.licensed.dll'\r\nTo resolve this, install the Chocolatey Licensed Extension package with\r\n `choco install chocolatey.extension`\r\n1.2.0\r\n\" to type \"System.Version\". Error: \"Version string portion was too short or too long.\"\r\nAt line:76 char:14\r\n+ ...        if ((Get-ChocolateyVersion -ChocoCommand $ChocoCommand) -lt [v ...\r\n+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : InvalidArgument: (:) [Get-ChocolateyVersion], RuntimeException\r\n    + FullyQualifiedErrorId : InvalidCastParseTargetInvocation,Get-ChocolateyVersion\r\n\r\nScriptStackTrace:\r\nat Get-ChocolateyVersion, <No file>: line 193\r\nat Get-ChocolateyPackage, <No file>: line 76\r\nat Get-ChocolateyPackageVersion<Begin>, <No file>: line 139\r\nat Install-Chocolatey, <No file>: line 1217\r\nat <ScriptBlock>, <No file>: line 139\r\n\r\nSystem.Management.Automation.RuntimeException: Cannot convert value \"Unable to load chocolatey.licensed assembly. Could not find a part of the path 'C:\\ProgramData\\chocolatey\\extensions\\chocolatey\\chocolatey.licensed.dll'.\r\nA valid Chocolatey license was found, but the chocolatey.licensed.dll assembly could not be loaded:\r\n  Could not load file or assembly 'chocolatey.licensed, Version=0.0.0.0, Culture=neutral, PublicKeyToken=' or one of its dependencies. The system cannot find the file specified.\r\nEnsure that the chocolatey.licensed.dll exists at the following path:\r\n 'C:\\ProgramData\\chocolatey\\extensions\\chocolatey\\chocolatey.licensed.dll'\r\nTo resolve this, install the Chocolatey Licensed Extension package with\r\n `choco install chocolatey.extension`\r\n1.2.0\r\n\" to type \"System.Version\". Error: \"Version string portion was too short or too long.\" ---> System.Management.Automation.PSInvalidCastException: Cannot convert value \"Unable to load chocolatey.licensed assembly. Could not find a part of the path 'C:\\ProgramData\\chocolatey\\extensions\\chocolatey\\chocolatey.licensed.dll'.\r\nA valid Chocolatey license was found, but the chocolatey.licensed.dll assembly could not be loaded:\r\n  Could not load file or assembly 'chocolatey.licensed, Version=0.0.0.0, Culture=neutral, PublicKeyToken=' or one of its dependencies. The system cannot find the file specified.\r\nEnsure that the chocolatey.licensed.dll exists at the following path:\r\n 'C:\\ProgramData\\chocolatey\\extensions\\chocolatey\\chocolatey.licensed.dll'\r\nTo resolve this, install the Chocolatey Licensed Extension package with\r\n `choco install chocolatey.extension`\r\n1.2.0\r\n\" to type \"System.Version\". Error: \"Version string portion was too short or too long.\" ---> System.ArgumentException: Version string portion was too short or too long.\r\n   at System.Version.VersionResult.SetFailure(ParseFailureKind failure, String argument)\r\n   at System.Version.TryParseVersion(String version, VersionResult& result)\r\n   at System.Version.Parse(String input)\r\n   --- End of inner exception stack trace ---\r\n   at System.Management.Automation.LanguagePrimitives.ConvertViaParseMethod.ConvertWithoutCulture(Object valueToConvert, Type resultType, Boolean recursion, PSObject originalValueToConvert, IFormatProvider formatProvider, TypeTable backupTable)\r\n   at CallSite.Target(Closure , CallSite , Object )\r\n   at System.Dynamic.UpdateDelegates.UpdateAndExecute1[T0,TRet](CallSite site, T0 arg0)\r\n   at System.Management.Automation.Interpreter.DynamicInstruction`2.Run(InterpretedFrame frame)\r\n   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)\r\n   --- End of inner exception stack trace ---\r\n   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)\r\n   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)\r\n   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)\r\n   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)\r\n   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)\r\n   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)\r\n   at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)\r\n   at System.Management.Automation.PSScriptCmdlet.DoEndProcessing()\r\n   at System.Management.Automation.CommandProcessorBase.Complete()", 

### What is Expected?

Both packages are installed without any issues

### How Did You Get This To Happen?

Specified v1.5.0 to be used:

collections:
  # Install a collection from Ansible Galaxy.
  - name: chocolatey.chocolatey
    version: 1.5.0
    source: https://galaxy.ansible.com

### System Details

Operating System: Windows Server 2022
Windows PowerShell version: 5.1.20348.1366
Chocolatey CLI Version: 1.2.0 and tested also 2.1.0
Chocolatey Licensed Extension version: 5.0.0 and tested also 6.1.0
Chocolatey License type: Business


### Installed Packages

```bash
N/A
```


### Output Log

```bash
Unable to load chocolatey.licensed assembly. Could not find a part of the path 'C:\ProgramData\chocolatey\extensions\chocolatey\chocolatey.licensed.dll'.
2023-07-03 14:37:06,365 6380 [ERROR] - A valid Chocolatey license was found, but the chocolatey.licensed.dll assembly could not be loaded:
  Could not load file or assembly 'chocolatey.licensed, Version=0.0.0.0, Culture=neutral, PublicKeyToken=' or one of its dependencies. The system cannot find the file specified.
Ensure that the chocolatey.licensed.dll exists at the following path:
 'C:\ProgramData\chocolatey\extensions\chocolatey\chocolatey.licensed.dll'
2023-07-03 14:37:06,365 6380 [WARN ] - To resolve this, install the Chocolatey Licensed Extension package with
 `choco install chocolatey.extension`
```


### Additional Context

_No response_

--- a/chocolatey/plugins/module_utils/Packages.psm1
+++ b/chocolatey/plugins/module_utils/Packages.psm1
@@ -193,6 +193,8 @@ function Get-ChocolateyVersion {
     # Also, if a license is installed, but the licensed extension is missing,
     # choco.exe output will contain an error which we need to ignore for the
     # purposes of determining the version of Chocolatey CLI.
+    # We're using the suggested regex for matching SemVer strings:
+    # https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
     $SemVerRegex = '(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?'
     if ($result.stdout -match $SemVerRegex) {
         ($script:ChocolateyVersion = [version]($matches[0] -replace '-.+$'))

--- a/chocolatey/plugins/module_utils/Packages.psm1
+++ b/chocolatey/plugins/module_utils/Packages.psm1
@@ -190,7 +190,17 @@ function Get-ChocolateyVersion {
 
     # Prerelease versions are not relevant for our purposes.
     # Stripping off any prerelease tag here gets us enough for what we need.
-    ($script:ChocolateyVersion = [version]($result.stdout -replace '-.+$'))
+    # Also, if a license is installed, but the licensed extension is missing,
+    # choco.exe output will contain an error which we need to ignore for the
+    # purposes of determining the version of Chocolatey CLI.
+    $SemVerRegex = '(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?'
+    if ($result.stdout -match $SemVerRegex) {
+        ($script:ChocolateyVersion = [version]($matches[0] -replace '-.+$'))
+    }
+    else {
+        $message = "Error getting version of Chocolatey CLI"
+        Assert-TaskFailed -Message $message -Command $command -CommandResult $result
+    }
 }
 
 function Get-CommonChocolateyArgument {

